ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/LedgerConfigProvisioner.scala

```diff
@@ -17,92 +16,86 @@ import com.daml.platform.configuration.LedgerConfiguration
-    extends AutoCloseable {
+object LedgerConfigProvisioner {
   private val logger = ContextualizedLogger.get(getClass)
 
-  private val closed: AtomicBoolean = new AtomicBoolean(false)
-
-  materializer.scheduleOnce(
-    config.initialConfigurationSubmitDelay.toNanos.nanos,
-    () => {
-      if (currentLedgerConfiguration.latestConfiguration.isEmpty && !closed.get)
-        submitInitialConfig(writeService)
-      ()
-    },
-  )
+  def owner(
+      ledgerConfiguration: LedgerConfiguration,
+      currentLedgerConfiguration: CurrentLedgerConfiguration,
+      writeService: state.WriteConfigService,
+      timeProvider: TimeProvider,
+      submissionIdGenerator: SubmissionIdGenerator,
+      scheduler: Scheduler,
+      servicesExecutionContext: ExecutionContext,
+  )(implicit
+      loggingContext: LoggingContext
+  ): ResourceOwner[Unit] = {
+    ResourceOwner
+      .forCancellable(() =>
+        scheduler.scheduleOnce(
+          ledgerConfiguration.initialConfigurationSubmitDelay.toNanos.nanos,
+          new Runnable {
+            override def run(): Unit = {
+              if (currentLedgerConfiguration.latestConfiguration.isEmpty)
+                submitInitialConfig(writeService, timeProvider, submissionIdGenerator)(
+                  ledgerConfiguration.initialConfiguration
+                )(servicesExecutionContext, loggingContext)
+              ()
+            }
+          },
+        )(servicesExecutionContext)
+      )
+      .map(_ => ())
+  }
 
   // There are several reasons why the change could be rejected:
   // - The participant is not authorized to set the configuration
   // - There already is a configuration, it just didn't appear in the index yet
   // This method therefore does not try to re-submit the initial configuration in case of failure.
-  private def submitInitialConfig(writeService: state.WriteConfigService): Future[Unit] = {
-    val submissionId = submissionIdGenerator.generate()
-    withEnrichedLoggingContext("submissionId" -> submissionId) { implicit loggingContext =>
-      logger.info(s"No ledger configuration found, submitting an initial configuration.")
-      DefaultTelemetry.runFutureInSpan(
-        SpanName.LedgerConfigProviderInitialConfig,
-        SpanKind.Internal,
-      ) { implicit telemetryContext =>
-        FutureConverters
-          .toScala(
-            writeService.submitConfiguration(
-              Timestamp.assertFromInstant(timeProvider.getCurrentTime.plusSeconds(60)),
-              submissionId,
-              config.initialConfiguration,
+  private def submitInitialConfig(
+      writeService: state.WriteConfigService,
+      timeProvider: TimeProvider,
+      submissionIdGenerator: SubmissionIdGenerator,
+  )(
+      initialConfiguration: Configuration
+  )(implicit executionContext: ExecutionContext, loggingContext: LoggingContext): Unit = {
+    Future.fromTry(Try(submissionIdGenerator.generate())).flatMap { submissionId =>
```

Comment:
When can this fail? Shouldn't we log something specific in that case?

Reply:
It's injected, so theoretically, whenever. That said, this might be a bit too belt-and-suspenders, because there's only one implementation and it calls `UUID.randomUUID()`, which (I think) should never fail, so I think I'll undo it.

Reply:
I reverted this change.

Reply:
I agree with the injected argument so I'd rather have logged the (theoretically possible) failure but it's a minor thing.